app.psgi

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/perl
+#
+# app.psgi
+#
+# Dreamwidth entrypoint for Plack.
+#
+# Authors:
+#      Mark Smith <mark@dreamwidth.org>
+#
+# Copyright (c) 2021 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself. For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+use strict;
+use v5.10;
+use Log::Log4perl;
+my $log = Log::Log4perl->get_logger(__PACKAGE__);
+
+BEGIN { require "$ENV{LJHOME}/cgi-bin/ljlib.pl"; }
+
+use Plack::Builder;
+
+use DW::Request::Plack;
+use DW::Routing;
+
+# Initial configuration that happens on server startup goes here, these things don't
+# change from request to request, so only put things here that never change
+BEGIN {
+    # If we're a DEV server, do some configuration
+    $^W = 1 if $LJ::IS_DEV_SERVER;
+
+    # Configure S2
+    S2::set_domain('LJ');
+
+    # Configure language library
+    # TODO: Why is this commented out here?
+    # my $lang = $LJ::DEFAULT_LANG || $LJ::LANGS[0];
+    # BML::set_language( $lang, \&LJ::Lang::get_text );
+
+    # Initialize random
+    # TODO: Make sure this is fired once per child if we're in preforking mode?
+    srand( LJ::urandom_int() );
+}
+
+my $app = sub {
+    my $r = DW::Request->get;
+
+    # Main request dispatch; this will determine what kind of request we're getting
+    # and then pass it to the appropriate handler. In the future, this should just
+    # be a call to DW::Routing and let it sort it out with all the controllers and
+    # such, but until then, we're having to dispatch between various generations
+    # of systems ourselves.
+    my $uri = $r->path;
+    $log->debug('Routing for URI: ', $uri);
+    if ( $uri =~ qr!^/api/v\d+/! ) {
```

Comment:
This only dispatches to API routes now, but this is where we need to add the rest of the dispatch engines... TT, BML, etc.

Although BML is probably a nightmare to pull off of Apache and maybe we should never support it here, which means Plack can't really be deployed until we finish the BML migration.